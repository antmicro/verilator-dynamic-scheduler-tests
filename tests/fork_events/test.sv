module t (/*AUTOARG*/
   // Inputs
   clk
   );
   input clk;
   event wake1;
   event wake2;
   event wake3;
   event finish;

   initial begin
      fork
         begin
            #1 $write("process 1\n");
            ->wake2;
            #1 @wake1 $write("process 1 again\n");
            ->wake3;
         end
         begin
            @wake2;
            #1 $write("process 2\n");
            ->wake3;
            @wake2 #1 $write("process 2 again\n");
            ->wake1;
         end
         begin
            @wake3 #1 $write("process 3\n");
            ->wake2;
            @wake3 $write("process 3 again\n");
            $write("*-* All Finished *-*\n");
            ->finish;
         end
         @finish $finish;
      join_none
   end
endmodule
