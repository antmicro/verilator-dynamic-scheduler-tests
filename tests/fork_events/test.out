process 1
process 2
process 3
process 2 again
process 1 again
process 3 again
*-* All Finished *-*
